File: 06-09-2024_Delete Nodes From Linked List Present in Array.java
import java.util.*;

class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }
}

public class Solution {
    public ListNode modifiedList(List<Integer> nums, ListNode head) {
        Set<Integer> set = new HashSet<>(nums);
        ListNode prev = null;
        ListNode curr = head;
        ListNode newHead = head;

        while (curr != null) {
            if (set.contains(curr.val)) {
                if (prev == null) {
                    newHead = curr.next;
                } else {
                    prev.next = curr.next;
                }
                // No direct memory management like delete in Java
                curr = (prev != null) ? prev.next : newHead;
            } else {
                prev = curr;
                curr = curr.next;
            }
        }
        return newHead;
    }
}

File: 06-09-2024_Delete Nodes From Linked List Present in Array.cpp
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) 
    {
        unordered_map<int,int> mp;
        for(int i=0;i<nums.size();i++)
            mp[nums[i]]=1;
        ListNode* prev=NULL;
        ListNode* curr=head;
        ListNode* newhead=head;
        while(curr!=NULL)
        {
            if(mp.find(curr->val)!=mp.end())
            {
                if(prev==NULL)
                  newhead=curr->next;
                else
                  prev->next=curr->next;
                delete(curr);
                if(prev)
                curr=prev->next;
                else 
                curr=newhead;
            }
            else{
                prev=curr;
                curr=curr->next;
            }
        }
        return newhead;
    }
};
